src/qengine/operators.cpp

```diff
@@ -172,167 +73,72 @@ void QEngineCPU::CINC(
         return;
     }
 
-    bitCapInt lengthPower = 1 << length;
-    bitCapInt lengthMask = lengthPower - 1;
+    bitCapInt lengthPower = 1U << length;
+    bitCapInt lengthMask = lengthPower - 1U;
     toAdd &= lengthMask;
-    if ((length > 0) && (toAdd > 0)) {
-        bitCapInt inOutMask = lengthMask << inOutStart;
-        bitCapInt otherMask = maxQPower - 1;
-
-        bitCapInt* controlPowers = new bitCapInt[controlLen];
-        bitCapInt controlMask = 0U;
-        for (bitLenInt i = 0; i < controlLen; i++) {
-            controlPowers[i] = 1U << controls[i];
-            controlMask |= controlPowers[i];
-        }
-        std::sort(controlPowers, controlPowers + controlLen);
-
-        otherMask ^= inOutMask | controlMask;
-
-        complex* nStateVec = AllocStateVec(maxQPower);
-        std::copy(stateVec, stateVec + maxQPower, nStateVec);
-
-        par_for_mask(0, maxQPower, controlPowers, controlLen, [&](const bitCapInt lcv, const int cpu) {
-            bitCapInt otherRes = lcv & otherMask;
-            bitCapInt inOutRes = lcv & inOutMask;
-            bitCapInt inOutInt = inOutRes >> inOutStart;
-            bitCapInt outInt = (inOutInt + toAdd) & lengthMask;
-            nStateVec[(outInt << inOutStart) | otherRes | controlMask] = stateVec[lcv | controlMask];
-        });
-
-        delete[] controlPowers;
-
-        ResetStateVec(nStateVec);
+    if ((length == 0U) || (toAdd == 0U)) {
+        return;
     }
-}
 
-/// Add BCD integer (without sign)
-void QEngineCPU::INCBCD(bitCapInt toAdd, bitLenInt inOutStart, bitLenInt length)
-{
-    int nibbleCount = length / 4;
-    if (nibbleCount * 4 != length) {
-        throw std::invalid_argument("BCD word bit length must be a multiple of 4.");
+    bitCapInt* controlPowers = new bitCapInt[controlLen];
+    bitCapInt controlMask = 0U;
+    for (bitLenInt i = 0U; i < controlLen; i++) {
```

Comment:
For my own edification: what happens if you don't supply the U?  Wouldn't the 0 get expanded to that len?

Reply:
`bitLenInt` and `bitCapInt` are guaranteed to be unsigned integral types. When we use implicitly signed literals in certain operations, like `1 << len`, the compiler can potentially get confused on the last bit of capacity in the type, for `bitCapInt`. (Something similar already happened with GCC on the ARM7.) Generally, wherever we're interacting with a `bitLenInt` or `bitCapInt`, we should default to unsigned literals, with the `U`.